### PR TITLE
Updating astrology to run safe-room if too injured to use telescope

### PR DESCRIPTION
**Pull request by fireballx777**

### CODE REVIEW

File: astrology.lic [hunk 1]
```diff
@@ -234,7 +234,12 @@ class Astrology
         bput('open my telescope', 'extend your telescope')
         return observe(body)
       end
-      result = bput('peer telescope', 'You peer aimlessly through your telescope', 'You see nothing regarding the future', 'Roundtime')
+      result = bput('peer telescope', 'You peer aimlessly through your telescope', 'You see nothing regarding the future', 'The pain is too much', 'Roundtime')
+	  if result =~ /The pain is too much/
+	    bput('stow telescope', 'you put', "That can't be picked up")
+		wait_for_script_to_complete('safe-room')
+		return observe(body)
```
Comment (jandersson) [inline astrology.lic]: Would this try to observe in the healers room?

File: astrology.lic [hunk 2]
```diff
@@ -234,7 +234,12 @@ class Astrology
         bput('open my telescope', 'extend your telescope')
         return observe(body)
       end
-      result = bput('peer telescope', 'You peer aimlessly through your telescope', 'You see nothing regarding the future', 'Roundtime')
+      result = bput('peer telescope', 'You peer aimlessly through your telescope', 'You see nothing regarding the future', 'The pain is too much', 'Roundtime')
+        if result =~ /The pain is too much/
+          bput('stow telescope', 'you put', "That can't be picked up")
+          wait_for_script_to_complete('safe-room')
+          return observe(body)
```
Comment (jandersson) [inline astrology.lic]: Not sure why my last commend is outdated, but i will try commenting again. Would this attempt observe in the healers room?

**Comment by rpherbig** [comment]
Will this fix #1810?

File: astrology.lic (hunk 2 above)
Comment (rpherbig) [inline astrology.lic]: This is how `pick.lic` handles this situation: https://github.com/rpherbig/dr-scripts/blob/master/pick.lic#L382-L384

Comment (jandersson) [inline astrology.lic]: Also do you want to recast buffs in case piercing gaze has expired while waiting to be healed?

Comment (fireballx777) [inline astrology.lic]: Updated with the logic from `pick.lic` to return to the starting room after running `safe-room.lic`.  Also re-running `dobuffs` to account for buffs dropping during the downtime.

**Comment by fireballx777** [comment]
Yes, this addresses #1810.

File: astrology.lic [hunk 3]
```diff
@@ -234,7 +234,16 @@ class Astrology
         bput('open my telescope', 'extend your telescope')
         return observe(body)
       end
-      result = bput('peer telescope', 'You peer aimlessly through your telescope', 'You see nothing regarding the future', 'Roundtime')
+      result = bput('peer telescope', 'You peer aimlessly through your telescope', 'You see nothing regarding the future', 'The pain is too much', 'Roundtime')
+        if result =~ /The pain is too much/
+          bput('stow telescope', 'you put', "That can't be picked up")
+          snapshot = Room.current.id
+          wait_for_script_to_complete('safe-room')
+          walk_to(snapshot)
```
Comment (rpherbig) [inline astrology.lic]: To use `walk_to` here, you need to add `common-travel` to `custom_require` and `include DRCT` above.

**Pull request merged**